test/e2e/index.test.ts

```diff
@@ -62,330 +62,490 @@ async function attemptGoto(page: playwright.Page, url: string): Promise<boolean>
-      await waitFor(async () => {
-        expect(await page.evaluate(() => document.activeElement?.textContent)).to.equal('Nike');
-        expect(await page.evaluate(() => document.activeElement?.getAttribute('role'))).to.equal(
-          'cell',
-        );
-      });
-      await page.keyboard.press('ArrowDown');
-      await waitFor(async () => {
-        expect(await page.evaluate(() => document.activeElement?.textContent)).to.equal('Adidas');
-        expect(await page.evaluate(() => document.activeElement?.getAttribute('role'))).to.equal(
-          'cell',
-        );
+
+      it('should implement the roving tabindex pattern', async () => {
+        await renderFixture('DataGrid/KeyboardNavigationFocus');
+
+        const button = page.locator('text=initial focus');
+        expect(await button.evaluate((node) => document.activeElement === node));
+
+        await page.keyboard.press('Tab');
+        await waitFor(async () => {
+          expect(await page.evaluate(() => document.activeElement?.textContent)).to.equal('brand');
+          expect(await page.evaluate(() => document.activeElement?.getAttribute('role'))).to.equal(
+            'columnheader',
+          );
+        });
+        await page.keyboard.press('ArrowDown');
+        await waitFor(async () => {
+          expect(await page.evaluate(() => document.activeElement?.textContent)).to.equal('Nike');
+          expect(await page.evaluate(() => document.activeElement?.getAttribute('role'))).to.equal(
+            'cell',
+          );
+        });
+        await page.keyboard.press('ArrowDown');
+        await waitFor(async () => {
+          expect(await page.evaluate(() => document.activeElement?.textContent)).to.equal('Adidas');
+          expect(await page.evaluate(() => document.activeElement?.getAttribute('role'))).to.equal(
+            'cell',
+          );
+        });
+        // Firefox insists on focusing the pagination wrapper first before the input.
+        // This might need looking into in more depth.
+        // Maybe it's a bug?
+        if (browserType.name() !== 'firefox') {
+          await page.keyboard.press('Tab');
+          await waitFor(async () => {
+            expect(await page.evaluate(() => document.activeElement?.textContent)).to.equal('100');
+            expect(
+              await page.evaluate(() => document.activeElement?.getAttribute('role')),
+            ).to.equal('button');
+          });
+          await page.keyboard.press('Shift+Tab');
+          await waitFor(async () => {
+            expect(await page.evaluate(() => document.activeElement?.textContent)).to.equal(
+              'Adidas',
+            );
+            expect(
+              await page.evaluate(() => document.activeElement?.getAttribute('role')),
+            ).to.equal('cell');
+          });
+          // WebKit does not want to return focus back to the button for some reason...
+          if (browserType.name() !== 'webkit') {
```

Comment:
By default, Safari doesn't focus on buttons 🙃
https://stackoverflow.com/a/67901876

Reply:
That's an "interesting" decision. 🤔 
Maybe it has its merit in increasing a form submission speed on a page where you'd have many buttons disturbing quick form navigation. 🤷 
But as always, the rule that "Safari is the new IE" still stands perfectly. 👍 😆

Reply:
https://github.com/microsoft/playwright/issues/5609#issuecomment-832684772